drivers/docker/config.go

```diff
@@ -96,10 +96,21 @@ func PluginLoader(opts map[string]string) (map[string]interface{}, error) {
 	conf["volumes"] = volConf
 
 	// capabilities
-	if v, ok := opts["docker.caps.whitelist"]; ok {
+	if v, ok := opts["docker.caps.allowlist"]; ok {
 		conf["allow_caps"] = strings.Split(v, ",")
 	}
 
+	// backward compatible configuration
+	if v, ok := opts["docker.caps.whitelist"]; ok {
+		vs := strings.Split(v, ",")
+		switch conf["allow_caps"].(type) {
+		case []string:
+			conf["allow_caps"] = append(conf["allow_caps"].([]string), vs...)
```

Comment:
I think merging with the previous config here would be a surprise behavior, especially without de-duplication.

We should set `conf["alloc_caps"]` first from `opts["docker.caps.whitelist"]`, and then completely override it with the value from `opts["docker.caps.allowlist"]`. That way the results if someone has both fields are a bit more obvious.

Reply:
Agreed.